src/main/java/com/zendesk/maxwell/MaxwellReplicator.java

```diff
@@ -315,56 +315,19 @@ protected BinlogEventV4 pollV4EventFromQueue() throws InterruptedException {
 	}
 
 
-	private void processQueryEvent(QueryEvent event) throws InvalidSchemaError, SQLException, IOException {
+	private void processQueryEvent(QueryEvent event) throws SchemaStoreException, InvalidSchemaError, SQLException {
 		// get charset of the alter event somehow? or just ignore it.
 		String dbName = event.getDatabaseName().toString();
 		String sql = event.getSql().toString();
+		BinlogPosition position = eventBinlogPosition(event);
 
-		List<SchemaChange> changes = SchemaChange.parse(dbName, sql);
-
-		if ( changes == null || changes.size() == 0 )
-			return;
-
-		ArrayList<ResolvedSchemaChange> resolvedSchemaChanges = new ArrayList<>();
-
-		Schema updatedSchema = getSchema();
-
-		for ( SchemaChange change : changes ) {
-			if ( !change.isBlacklisted(this.filter) ) {
-				ResolvedSchemaChange resolved = change.resolve(updatedSchema);
-				if ( resolved != null ) {
-					resolved.apply(updatedSchema);
-
-					resolvedSchemaChanges.add(resolved);
-				}
-			} else {
-				LOGGER.debug("ignoring blacklisted schema change");
-			}
-		}
-
-		if ( resolvedSchemaChanges.size() > 0 ) {
-			BinlogPosition p = eventBinlogPosition(event);
-			LOGGER.info("storing schema @" + p + " after applying \"" + sql.replace('\n', ' ') + "\"");
-
-			saveSchema(updatedSchema, resolvedSchemaChanges, p);
-		}
-	}
-
-	private void saveSchema(Schema updatedSchema, List<ResolvedSchemaChange> changes, BinlogPosition p) throws SQLException {
+		schemaStore.processSQL(sql, dbName, position);
 		tableCache.clear();
-
-		if ( !this.context.getReplayMode() ) {
-			try (Connection c = this.context.getMaxwellConnection()) {
-				this.savedSchema = this.savedSchema.createDerivedSchema(updatedSchema, p, changes);
-				this.savedSchema.save(c);
-			}
-
-			this.producer.writePosition(p);
-		}
+		this.producer.writePosition(position);
```

Comment:
should be ok.  in replay mode the position store is a `ReadOnlyMysqlPositionStore`